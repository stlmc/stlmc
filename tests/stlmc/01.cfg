common {
    time-bound = 25
    bound = 2
    solver = "yices"
    verbose = "true"   # print verbose messages
}

yices   {
    logic = "QF_NRA"
}
