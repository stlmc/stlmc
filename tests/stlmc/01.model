real b1;
real b2;

[0, 10] g1;
[0, 10] g2;
[-30, 30] d2;
[-30, 30] d1;

{
    mode:
        b1 = 1;
        b2 = 1;
    inv:
 	g1 > 2;
	g2 > 2;
    flow:
	d1(t) = 0.1 * (1 - 2 * t + 2 * t * t) + 0.5 + d1(0);
	d2(t) = 0.1 * (1 - 2 * t + 2 * t * t) + 0.5 + d2(0);
	g1(t) = -0.5 * t + g1(0);
	g2(t) = -0.5 * t + g2(0);
    jump:
        g1 > 1 =>
          (and (b1' = 1) (b2' = 2) (d1' = d1) (d2' = (d2 - 0.122 * g2)) (g1' = g1) (g2' = g2));
        g2 > 1 =>
          (and (b1' = 2) (b2' = 1) (d1' = d1)  (d2' = (d2 - 0.122 * g2)) (g1' = g1) (g2' = g2));
}
{
    mode:
        b1 = 1 ;
        b2 = 2 ;
    inv:
	g1 > 0.5;
	d2 > 1;
    flow:
	d1(t) = 0.1 * (1 - 2 * t + 2 * t * t) + 1 + d1(0);
	d2(t) = -0.1 * (1 - 2 * t + 2 * t * t) + d2(0);
	g1(t) = g1(0) - t;
	g2(t) = g2(0);
    jump:
        g2 > 1 =>
          (and (b1' = 2) (b2' = 1) (d1' = d1) (d2' = (d2 - 0.122 * g2)) (g1' = g1) (g2' = g2));
        g1 <=1 =>
          (and (b1' = 3) (b2' = 1) (d1' = d1) (d2' = (d2 - 0.122 * g2)) (g1' = g1) (g2' = g2));
}


init:
(and
    (b1 = 1) (b2 = 1) (8.4 <= g1) (g1 <= 8.5) (0 <= d1) (d1 <= 0.1)
    (7.4 <= g2) (g2 <= 7.5) (0 <= d2) (d2 <= 0.1)) ;

proposition:

# bound: 10, timebound: 25
goal:
[f1]: <>[6, 10] (d2 >= 4); # threshold: 2.5

