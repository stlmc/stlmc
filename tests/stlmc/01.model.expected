size: 396
result: true